src/realm/list.cpp

```diff
@@ -678,6 +690,84 @@ void Lst<Mixed>::add_index(Path& path, Index index) const
     path.emplace_back(ndx);
 }
 
+bool Lst<Mixed>::nullify(ObjLink link)
+{
+    size_t ndx = find_first(link);
```

Comment:
It strikes me as somewhat of a problem if `find_first` isn't working anymore, though I'm not sure where we'd see the side effects of that. Would it make sense to make `find_first` return a path to the index in a nested collection if applicable?

Reply:
I am sure it will be a problem if `find_first` is not working, but I am not sure what you are aiming at here. I think `find_first` should only work at current list. It would be double work if you would first try to find the link in a nested collection and then recreate the same collection using the path returned.

Reply:
I was pointing out that users may expect `Lst<Mixed>::find_first(x)` to return a match even if the value is found in a nested collection. But, comparing with how linklists work, now I have come to the same conclusion as you; that it only makes sense to search the top level.